File: Easy/1.Two Sum/01_attempt.py
#================================
# Working process:
#   1. Iterate through nums
#   2. Iterate through nums once again from where we left off
#   3. Check if sum of those two elements of nums are equal to target
# Issue: Way too long Runtime
#================================


class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        for i in range(len(nums)):
            for j in range(i+1,len(nums)):
                if nums[i]+nums[j] == target:
                    return[i,j]

File: Easy/1.Two Sum/01_attempt.cpp

#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> pairs;
        int comp;

        for(int i{0}; i<nums.size(); ++i){
            comp = target - nums[i];
            if(pairs.contains(comp)){
                return std::vector<int> {pairs[comp],i};
            }
            else{
                pairs[nums[i]] = i;
            }
        }

        return std::vector<int> {};
    }
};